### PR TITLE
Use case dg

### PR DESCRIPTION
**Pull request by prit3010**
Add use cases in dg

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -552,73 +553,251 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+  
+- 3c. Fields are empty.
+
+  - 3c1. Coydir shows an error message .
 
     Use case resumes at step 2.
 
-**Use case: Find details of a person**
+
+**Use case: UC03 - Find details of a employee**
```
Comment (kevinchangjk) [inline docs/DeveloperGuide.md:600]: "Find details of an employee".

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -552,73 +553,251 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
     Use case ends.
 
-**Use case: View details of a person**
+- 2b. The fields in the query are not in the correct format.
+
+  - 2b1. Coydir shows error message informing user which field has invalid input and why it is invalid.
+  
+  Use case ends.
+
+**Use case: UC04 - View details of a employee**
```
Comment (kevinchangjk) [inline docs/DeveloperGuide.md:623]: Same here.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -552,73 +553,251 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-- 3a. The given index is invalid.
+- 3a. The given ID is invalid.
 
   - 3a1. Coydir shows an error message.
 
   Use case resumes at step 2.
 
-**Use case: Edit details of a person**
+**Use case: UC02 - Edit details of a employee**
```
Comment (kevinchangjk) [inline docs/DeveloperGuide.md:562]: Same here.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -494,53 +494,54 @@ _Note_:
...
+| `* * `  | HRE                                        | view the claims (transport claims, food claims etc) that employees make | compensate them back.                                                                             |
+| `* * `  | expert user                                | customizable shortcuts                                                  | run frequently used commands faster.                                                              |
 
 
 ### Use cases
 
 (For all use cases below, the **System** is the `Coydir` and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case: UC01 - Delete a person**
```
Comment (kevinchangjk) [inline docs/DeveloperGuide.md:537]: Want change to "employee"?

**Review by kevinchangjk — COMMENTED** [review]
Minor nits.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -552,73 +553,251 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1. User requests to view details of department.
+2. Coydir shows details of requested department.
+
+**Extensions**
+- 2a. Department does not exist.
+  - 2a1. Coydir shows an error message.
+  
+  Use case ends.
+
+**Use case: UC10 Rate the performance of an employee**
```
Comment (Kok-je) [inline docs/DeveloperGuide.md:800]: missing hyphen here: "UC10 - Rate the performance of an employee"

**Review by Kok-je — CHANGES_REQUESTED** [review]
Fix error with grammar

**Review by Kok-je — COMMENTED** [review]
LGTM

**Review by Kok-je — APPROVED** [review]
LGTM part 2

**Pull request merged**